src/dbnode/persist/fs/seek_manager.go

```diff
@@ -315,8 +362,19 @@ func (m *seekerManager) Borrow(shard uint32, start time.Time) (ConcurrentDataFil
 	return availableSeeker.seeker, nil
 }
 
+func (m *seekerManager) shardExistsWithLock(shard uint32) bool {
+	_, err := m.shardSet.LookupStateByID(shard)
+	// NB(bodu): LookupStateByID returns ErrInvalidShardID when shard
+	// does not exist in the shard map which means the shard is not available.
+	return err != sharding.ErrInvalidShardID
```

Comment:
Maybe we just check `err == nil`? Seems like any error looking up the shard is probably ok.

Reply:
Just want to make sure if the calling code changes the error returned but leaves the var there we still behave ok.